src/Neuron.js

```diff
@@ -70,40 +69,27 @@ class Neuron {
   /**
    * Train the Neuron to output the `targetOutput`.  If a `targetOutput`
    * is not provided, the Neuron will train itself to minimize the error
-   * of the Neurons from its outgoing connections.
+   * of the Neurons at its outgoing connections.
    * @param {number} [targetOutput] - Manually set the target output.error.
    */
   train(targetOutput) {
-    const inputDerivative = this.activation.prime(this.input)
+    // input and bias neurons have no incoming connections to update
+    if (this.isInput() || this.isBias) return
```

Comment:
Weights are now updated from a Neuron's incoming connection fan (ie from the Neuron at the target end of the connection).  This is also where the `delta` is calculated.

Since all calculation and weight update is now done from the output side of the connection, we have no need to invoke `train()` on input and bias neurons.  They have no incoming connections to train.  Because of this, they do not need to calculate deltas since there are no upstream Neurons to use the deltas.

This greatly simplified the logic in this method.  Also, by moving from the Delta Rule to regular backpropagation we increased our learning efficiency by 3.333x.